.github/workflows/release-drafter.yaml

```diff
@@ -31,12 +31,69 @@ jobs:
     env:
       RELEASE_VERSION: ${{ needs.release-version.outputs.version }}
     steps:
+      - name: Generate token
+        id: generate_token
+        uses: tibdex/github-app-token@v2
+        with:
+          app_id: ${{ secrets.OCMBOT_APP_ID }}
+          private_key: ${{ secrets.OCMBOT_PRIV_KEY }}
       - name: Checkout
         uses: actions/checkout@v4
-      - name: Drafter
-        uses: release-drafter/release-drafter@v6
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
-          tag: ${{env.RELEASE_VERSION }}
-          version: ${{env.RELEASE_VERSION }}
\ No newline at end of file
+          fetch-depth: 0
+          fetch-tags: true
+      - name: Setup Release with gh
+        env:
+          REF: ${{ github.ref }}
+          REPO: ${{ github.repository }}
+          GH_TOKEN: ${{ steps.generate_token.outputs.token }}
+        run: |
+          git fetch --tags
+          # generate the release notes based on the last previous tag.
+          # do not match for any latest tag but only a valid semver tag.
+          # also do not match for the cutoff tag for the release branch.
+          # Example: If we are in releases/v0.18, we don't want to match v0.18 as the cutoff tag.
+          # Instead we want to match the one before that, which would be v0.17.
+          # That would generate us the notes from
+          # v0.17 to HEAD, which is what we want.
+          previous_tag=$(git describe HEAD --abbrev=0 --tags --match "v*" --exclude "*latest*" --exclude "${REF#releases/}")
+          
+          if [[ -z $previous_tag ]]; then
+            echo "No previous tag found, cannot generate release notes"
+            exit 1
+          fi
+          
+          echo "Generating release notes for ${{env.RELEASE_VERSION}} starting from ${previous_tag} to HEAD"
+          
+          notes=$(\
+             gh api  \
+               --method POST \
+               -H "Accept: application/vnd.github+json" \
+               -H "X-GitHub-Api-Version: 2022-11-28" \
+               /repos/${REPO}/releases/generate-notes \
+               -f "tag_name=${{env.RELEASE_VERSION}}" \
+               -f "target_commitish=${{ github.ref }}" \
+               -f "previous_tag_name=${previous_tag}" \
+               -f "configuration_file_path=.github/release.yml" \
```

Comment:
When we do have the option to specify which config to use... can we then maybe place the config under `.github/config`?

Reply:
My original thought was that I would like to keep the default location as marked in the official documentation but im also fine to adopt it. WDYT?

Reply:
@frewilhelm ?

Reply:
adjusted. changed my mind :D